src/sagas/systemIntakeSaga.ts

```diff
@@ -25,11 +31,21 @@ function postSystemIntakeRequest(formData: SystemIntakeForm) {
   return axios.post(`${process.env.REACT_APP_API_ADDRESS}/system_intake`, data);
 }
 
-function* createSystemIntake(action: Action<any>) {
+function* createSystemIntake(
+  action: Action<{ formData: SystemIntakeForm; options?: OptionalSagaOptions }>
```

Comment:
Trying out something new here.

Instead of only having the data as the payload, I'm passing optional options as a part of the payload. This means it will look like:
```
{
  type: 'SOME_ACTION_TYPE'
  payload: {
    formData: {},
    options: {}
  }
}
```

This allows us to pass optional functions, parameters, values that might be needed. In this case, we need a `handleOnSuccess` to invoke after a successful API request. In the future, we might need `handleOnFailure` or even `handleOnComplete` for after the flow is fully complete.